geopyspark/rasterio/read.py

```diff
@@ -0,0 +1,123 @@
+import os
+import math
+import rasterio
+import geopyspark as gps
+import numpy as np
+
+
+# On driver
+if "GDAL_DATA" in os.environ:
+    __GDAL_DATA = os.environ["GDAL_DATA"]
+else:
+    __GDAL_DATA = None
+
+
+def dataset_to_proj4(dataset):
+    from osgeo import osr
```

Comment:
Similar to the `rasterio` import as we don't require `osgeo` to be installed to use GPS
```python
try:
    from osgeo import osr
except ImportError:
    raise ImportError("osgeo must be installed in order to use the dataset_to_proj4 function")
```